lib/mongoid_rails_migrations/mongoid_ext/railties/database.rake

```diff
@@ -1,17 +1,23 @@
 namespace :db do
-  unless Rake::Task.task_defined?("db:drop")
-    desc 'Drops all the collections for the database for the current Rails.env'
-    task :drop => :environment do
-      Mongoid.master.collections.each {|col| col.drop_indexes && col.drop unless ['system.indexes', 'system.users'].include?(col.name) }
+  if Rake::Task.task_defined?("db:drop")
```

Comment:
Why this change?

Reply:
To provide a way to drop a specific database.
Also the `Mongoid.master` method does not exist anymore.

Reply:
Mm but why would we need to `clear` existing task more than before?

Also to get the client database you can use `Mongoid.current_client.database` instead of going through the DataMigration collection.

Reply:
> Mm but why would we need to clear existing task more than before?
If we don't do this, it will append this task to the existing one, we can keep it but it could drop the database twice what is particularly useless.

> Also to get the client database you can use Mongoid.current_client.database instead of going through the DataMigration collection.

Good point :+1:

Reply:
Which existing one? from Rails? the gem was not doing this before so it was not working and using ActiveRecord task?

Reply:
I don't know, probably the one defined by `ActiveRecord` yes, you can see in the diff that before we were defining the task only if it was not defined:
```ruby
unless Rake::Task.task_defined?("db:drop")
  desc 'Drops all the collections for the database for the current Rails.env'
  task :drop => :environment do
    Mongoid.master.collections.each {|col| col.drop_indexes && col.drop unless ['system.indexes', 'system.users'].include?(col.name) }
  end
end


```

Reply:
Yes so the question still stands: why did you change this if it was like this before?

Reply:
I've already answered to this question:
> If we don't do this, it will append this task to the existing one, we can keep it but it could drop the database twice what is particularly useless.

Reply:
Maybe I'm missing something but it still does not answer my question: why do we need this NOW and not BEFORE?

Reply:
I think I've already answered to this question above but I will reformulate with more deails:

**BEFORE**
The task was already present in the mongoid gem:
```ruby
desc "Drops the default client database"
  task :drop => :environment do
    ::Mongoid::Clients.default.database.drop
end
```
This is deleting the database for the default client **ONLY**.

**NOW**
We're adding multi shards so we propose to pass the targeted database as parameter (in the env variable) like:
```
MONGOID_CLIENT_NAME=shard1 rails db:drop
```
We are now removing the old task **BECAUSE** if we don't this both task will be called and that's not what we want. It was not necessary before the default task was already doing the expected job.

Reply:
Ok thanks, that's better indeed :+1:
So this means the previous task was useless though (as already defined in Mongoid)? Or maybe it was for very old versions of mongoid which did not define it?

Reply:
Most probably yes, FWICS the database management tasks appeared only in 2014 in the mongoid gem: https://github.com/mongodb/mongoid/commits/master/lib/mongoid/tasks/database.rake before it was using the ActiveRecord task while this gem added support database drop from the beginning in 2010: https://github.com/adacosta/mongoid_rails_migrations/commit/39e5004666cf895a8fe15051a76fcd6828266740#diff-b87e05cc243468dabc4100dbfdd30f248ac332282dca6bb7655eef9a0ee303aa